client/scripts/views/modals/webhook_settings_modal.ts

```diff
@@ -0,0 +1,134 @@
+import 'modals/webhook_settings_modal.scss';
+
+import m from 'mithril';
+import $ from 'jquery';
+import app from 'state';
+import { Button, List, ListItem, Checkbox } from 'construct-ui';
+import { Webhook } from 'models';
+import { NotificationCategories } from 'types';
+import { EdgewareChainNotificationTypes, KusamaChainNotificationTypes, KulupuChainNotificationTypes, PolkdotChainNotificationTypes } from 'helpers/chain_notification_types';
+import { symbols } from 'helpers';
+
+interface IAttrs {
+    webhook: Webhook;
+    updateSuccessCallback: Function;
+}
+
+interface IState {
+    selectedCategories: string[];
+    // webhook: Webhook;
+}
+const forumNotificationTypes = [
+    NotificationCategories.NewThread,
+    NotificationCategories.NewComment,
+    NotificationCategories.NewReaction,
+]
+
+const WebhookSettingsModal: m.Component<IAttrs, IState> = {
+  oninit: (vnode) => {
+    vnode.state.selectedCategories = [];
+    vnode.attrs.webhook.categories.forEach((v) => vnode.state.selectedCategories.push(v));
+  },
+  view: (vnode: m.VnodeDOM<IAttrs, IState>) => {
```

Comment:
Prefer to avoid using m.VnodeDOM. Better to have no typing at all in `view`, in which case IAttrs and IState's typings will be inferred from the parent m.Component